sirepo/package_data/static/js/sirepo-components.js

```diff
@@ -1842,6 +1842,39 @@ SIREPO.app.directive('simplePanel', function(appState, panelState) {
     };
 });
 
+SIREPO.app.directive('simStateProgressBar', function(appState) {
+    return {
+        restrict: 'A',
+        scope: {
+            simState: '<',
+            defaultClass: '@',
+        },
+        template: `
+            <div class="progress">
+              <div class="progress-bar {{ class() }}" role="progressbar" aria-valuenow="{{ percentComplete() }}" aria-valuemin="0" aria-valuemax="100" data-ng-attr-style="width: {{ percentComplete() || 100 }}%"></div>
+            </div>
+        `,
+        controller: function($scope) {
+            $scope.class = () => {
+                if ($scope.defaultClass) {
+                    return $scope.defaultClass;
+                }
+                if ($scope.simState.isInitializing()) {
+                    return 'progress-bar-striped active';
+                }
+                return '';
+            };
+
+            $scope.percentComplete = () => {
+                if ($scope.simState && $scope.simState.getPercentComplete) {
```

Comment:
Is there a case where we have simState but no getPercentComplete function?

Reply:
When you open sirepo.com/srw#/simulations simState is defined, but getPercentComplete is not
![Screenshot 2023-11-16 at 1 27 47 PM](https://github.com/radiasoft/sirepo/assets/49498231/634d4f91-f614-4941-b186-95d0839e9e66)

Reply:
Why is the progress bar being rendered on the simulation list page? That seems strange.

Reply:
The appFooter in srw.js has a data-download-status and data-download-status has a sim-state-progress-bar

Reply:
Ah, thanks. `download-status` is a strange directive. It has to [fake out simState](https://github.com/radiasoft/sirepo/blob/19c2e6884f4b13a31b70ba15f90125f5fdcd4b17/sirepo/package_data/static/js/srw.js#L920) and then [sets it later](https://github.com/radiasoft/sirepo/blob/19c2e6884f4b13a31b70ba15f90125f5fdcd4b17/sirepo/package_data/static/js/sirepo-components.js#L3442). We should put the check for simState being valid inside of `download-status` that way we don't have to muddle all code below it with checks.

`<div data-ng-if="angular.equals(simState, {})" class="modal fade" id="sr-download-status" tabindex="-1" role="dialog">`